addons/l10n_ke_edi_tremol/models/account_move.py

```diff
@@ -142,8 +142,13 @@ def is_candidate(discount_line, other_line):
         # The device expects all monetary values in Kenyan Shillings
         if self.currency_id == self.company_id.currency_id:
             currency_rate = 1
+        elif (self.move_type != 'out_invoice'):
+            reversed_lines = self.reversed_entry_id.invoice_line_ids.filtered(
+                lambda l: not l.display_type and l.quantity and l.price_total
+            )
+            currency_rate = sum(abs(line.balance) for line in reversed_lines) / sum(abs(line.price_subtotal) for line in reversed_lines)
```

Comment:
We could also use the fields on the credit note itself: amount_...._signed and amount...